### PR TITLE
REST Modding Example

### PR DESCRIPTION
**Pull request by JulianMa**
This is a basic working example for an extension of the Eco WebServer Rest API.
This will be possible in a future Version of Eco.

### CODE REVIEW

**Comment by JulianMa** [comment]
I added the README. 
It should be notes, that this feature will only be available, after it is Supported by the EcoServer:
https://github.com/StrangeLoopGames/Eco/pull/10888

**Review by thetestgame — APPROVED** [review]

File: Examples/ExtendRestApi/Server/HelloWorldController.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{
+     /// <summary>
+    /// Example Controller to extend the Eco WebServer with a custom Endpoint.
+    /// More details are available at the official ASP.NET Core MVC Documentation from Microsoft.
+    /// </summary>
+    [AllowAnonymous, Route("api/v1/helloworld")]
+    public class HelloWorldController : Controller
+    {
+        [HttpGet("")]
+        public String getAll()
```
Comment (redwyre) [inline Examples/ExtendRestApi/Server/HelloWorldController.cs:15]: Should just be called `Get()`

File: Examples/ExtendRestApi/Server/HelloWorldController.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+using System;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Eco.WebServer.Web.Controllers
```
Comment (redwyre) [inline Examples/ExtendRestApi/Server/HelloWorldController.cs:5]: Don't use the Eco namespace, use your mods namespace. This makes it hard to tell if errors are coming from Eco or mods.

**Review by redwyre — CHANGES_REQUESTED** [review]

File: Examples/ExtendRestApi/Server/HelloWorldController.cs (hunk 2 above)
Comment (JulianMa) [reply Examples/ExtendRestApi/Server/HelloWorldController.cs]: I fixed both.

**Review by redwyre — APPROVED** [review]

**Pull request merged**